sklearn/ensemble/gradient_boosting_cv.py

```diff
@@ -0,0 +1,804 @@
+            score_tuple = _CVScoreTuple(params, mean, scores)
+            grid_scores.append(_CVScoreTuple(params, mean, scores))
+
+            if mean > best_mean:
+                best_tuple = score_tuple
+                best_params = params
+
+        self.grid_scores_ = grid_scores
+        self.best_score_tuple_ = best_tuple
+        self.best_params_ = best_params
+
+        if self.refit:
+            self.best_params_['random_state'] = self.random_state
+            gb = self._estimator_class(**self.best_params_)
+            gb.fit(X, y)
+            self.best_estimator_ = gb
+
+        return self
+
+    def predict(self, X):
+        """Call `predict` on the best estimator.
+
+        Parameters
+        ----------
+        X : array-like of shape = [n_samples, n_features]
+            The input samples.
+        Returns
+        -------
+        y: array of shape = ["n_samples]
+            The predicted values.
+        """
+        return self.best_estimator_.predict(X)
+
+    def _get_params(self):
+        "Return parametrs to iterate over for underlying gradient boosting"
+
+        params = {
+            'loss': self.loss,
+            'learning_rate': self.learning_rate,
+            'subsample': self.subsample,
+            'min_samples_split': self.min_samples_split,
+            'min_samples_leaf': self.min_samples_leaf,
+            'min_weight_fraction_leaf': self.min_weight_fraction_leaf,
+            'max_depth': self.max_depth,
+            'init': self.init,
+            'max_features': self.max_features,
+            'max_leaf_nodes': self.max_leaf_nodes,
+            'presort': self.presort,
+            'random_state': self.random_state,
+        }
+
+        # Pre processing to ensure every parameter is a list
+        # which `ParameterGrid` can iterate over
+        for key, value in params.items():
+            params[key] = np.atleast_1d(value)
+
+        return params
+
+
+def _fit_single_param(estimator, X, y, train, validation, params, stop_rounds,
+                      max_iter, scoring, score_precision):
+    """ Fit a single estimator till stopping criteria is reached.
+
+    Parameters
+    ---------
+    estimator : GradientBoostingRegressor or GradientBoostingClassifier
+        The class of the model to instantiate.
+
+    X : array-like, shape = [n_samples, n_features]
+        Training vectors, where n_samples is the number of samples
+        and n_features is the number of features.
+
+    y : array-like, shape = [n_samples]
+        Target values (integers in classification, real numbers in
+        regression)
+        For classification, labels must correspond to classes.
+
+    train : array-like
+        The indices of the training samples.
+
+    validation : array-like
+        The indices of the validation samples.
+
+    params : dict
+        The parameters to instantiate the Gradient Boosting model with
+
+    stop_rounds : int
```

Comment:
at least be consistent with `n_iter_no_change` in your class `GradientBoostingRegressorCV`